### PR TITLE
[Don't Review] Fix torch.norm with 2 dims

### LINKED ISSUE #37323: torch.norm is numerically unstable at zero for multidim reductions
Note the inconsistent gradient computation:

```
import torch
a = torch.zeros(3, 3, 3, requires_grad = True)
print(torch.autograd.grad(a.norm(dim = 1).sum(), (a,))[0])
print(torch.autograd.grad(a.norm(dim = (1, 2)).sum(), (a,))[0])
print(torch.__version__)
```

```
tensor([[[0., 0., 0.],
         [0., 0., 0.],
         [0., 0., 0.]],

        [[0., 0., 0.],
         [0., 0., 0.],
         [0., 0., 0.]],

        [[0., 0., 0.],
         [0., 0., 0.],
         [0., 0., 0.]]])
tensor([[[nan, nan, nan],
         [nan, nan, nan],
         [nan, nan, nan]],

        [[nan, nan, nan],
         [nan, nan, nan],
         [nan, nan, nan]],

        [[nan, nan, nan],
         [nan, nan, nan],
         [nan, nan, nan]]])
1.6.0.dev20200417
```

Relevant code: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/LinearAlgebra.cpp#L558

Relevant discussion: https://github.com/pytorch/pytorch/issues/37272#issuecomment-619405043

cc @ezyang @SsnL @albanD @zou3519 @gqchen @ngimel

### PR DESCRIPTION
**Pull request by mrshenli**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43996 [Don't Review] Fix torch.norm with 2 dims**

fixes #37323

### CODE REVIEW

**Comment by mruberry** [comment]
Not a review, but note that we'll be deprecating torch.norm in favor of torch.linalg.norm. So if this change would break BC you can apply it only to torch.linalg.norm without worry (since it's so new).
